### PR TITLE
Ensure HttpClient.Get*Async helpers wrap IOExceptions

### PR DESCRIPTION
**Pull request by stephentoub**
IOExceptions may emerge from the underlying response streams, but GetString/ByteArrayAsync should throw a wrapper HttpRequestException in this case.

Fixes https://github.com/dotnet/corefx/issues/39118
cc: @davidsh, @geoffkizer, @scalablecory, @wfurt

### CODE REVIEW

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs [hunk 1]
```diff
@@ -745,6 +745,34 @@ public async Task GetAsync_InvalidHeaderNameValue_ThrowsHttpRequestException(str
...
+        [InlineData(true, true)]
+        public async Task GetAsync_IncompleteData_ThrowsHttpRequestException(bool failDuringHeaders, bool getString)
+        {
+            if (IsWinHttpHandler)
+            {
+                // [ActiveIssue(39136)]
+                return;
+            }
+
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
```
Comment (geoffkizer) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Would be nice to make this work for HTTP2 as well. I'm fine if you want to defer that, though.

Comment (stephentoub) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Ah, right, will fix tomorrow.

Comment (stephentoub) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Actually, doesn't look like we have anything on the GenericLoopbackServer today that enables this, at least for failDuringHeaders=true.  We can follow-up on this separately, along with the vast majority of other tests in this file that are 1.1-specific.  The good news is this logic is entirely contained in HttpClient itself, which is agnostic as to HTTP version.

Comment (geoffkizer) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Think of this as an opportunity to improve GenericLoopbackServer :)

(seriously though, i'm fine if these are 1.1 specific for now)

**Pull request merged**